pkg/control/sidecarcontrol/util.go

```diff
@@ -213,6 +213,32 @@ func GetPodsSortFunc(pods []*corev1.Pod, waitUpdateIndexes []int) func(i, j int)
 	}
 }
 
+func IsPodInjectedSidecarSet(pod *corev1.Pod, sidecarSet *appsv1alpha1.SidecarSet) bool {
+	sidecarSetNameStr, ok := pod.Annotations[SidecarSetListAnnotation]
+	if !ok || len(sidecarSetNameStr) == 0 {
+		return false
+	}
+	sidecarSetNames := sets.NewString(strings.Split(sidecarSetNameStr, ",")...)
+	return sidecarSetNames.Has(sidecarSet.Name)
+}
+
+func IsPodConsistentWithSidecarSet(pod *corev1.Pod, sidecarSet *appsv1alpha1.SidecarSet) bool {
+	for i := range sidecarSet.Spec.Containers {
+		sc := &sidecarSet.Spec.Containers[i]
+		var container *corev1.Container
+		switch sc.UpgradeStrategy.UpgradeType {
+		case appsv1alpha1.SidecarContainerHotUpgrade:
+			container = util.GetContainer(fmt.Sprintf("%v-1", sc.Name), pod)
```

Comment:
also check for the backup container?
			container2 = util.GetContainer(fmt.Sprintf("%v-2", sc.Name), pod)

Reply:
done